### PR TITLE
migrate SQL Server Schemas to NVARCHAR

### PR DESCRIPTION
**Pull request by leviramsey**
References #650

### CODE REVIEW

**Comment by leviramsey** [comment]
Re: docs, I'm not sure where a "if using a journal/snapshot store created with Akka Persistence versions 5.0.4 and earlier, it is advised to disable the JDBC client's sending strings as unicode" note should go.

**Comment by leviramsey** [comment]
Thoughts on restructuring the docs along the lines of the Cassandra plugin with notes for the different DBs?

**Comment by octonato** [comment]
About the docs, I think we should add it to the migration guide. 

Now it's a good moment to do it. We have another [PR](https://github.com/akka/akka-persistence-jdbc/pull/648), less critical, that will also require a migration guide. 

And we didn't documented the migration tool that was added in #603. 

Maybe we should go for 5.1.0 because of the DB changes.

**Comment by leviramsey** [comment]
We should document a recommendation if using the old (`VARCHAR`-based) SQL Server schemas to use the `;sendStringParametersAsUnicode=false` connection parameter.

**Comment by leviramsey** [comment]
Actually I guess we can support offline migration of data from an old journal/snapshot-store to a new journal/snapshot-store, but I think it would be generally preferable from the user perspective to use the old schema and configure JDBC to work properly with the old schema.

That said, the migration tool would support the "turns out, we need unicode persistence IDs or tags" use case.

**Comment by patriknw** [comment]
@leviramsey would it be possible to push this forward?

**Comment by leviramsey** [comment]
I guess the question is if we want to implement a migration tool (which honestly is not something I'm confident on being able to implement any time soon).

As it is, this should just be mergeable (it requires no code changes, AFAIK), perhaps with docs around `;sendStringParametersAsUnicode=false` for use with previous versions of the schema.

File: core/src/main/resources/schema/sqlserver/sqlserver-create-schema-varchar.sql [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*
+Akka Persistence JDBC versions from 5.0.0 through 5.0.4 used this schema.  The only difference from the
```
Comment (patriknw) [inline core/src/main/resources/schema/sqlserver/sqlserver-create-schema-varchar.sql:2]: also 5.1.0

**Review by patriknw — APPROVED** [review]
Looking good. We don't need to provide a migration tool. DBAs (or any developer with some google skills) can do that with SQL.

Only a small addition of the version numbers.

**Review by johanandren — APPROVED** [review]

**Pull request merged**